Problem with quotes in test.py

Hi,

for the thorough testing on CI I need to have possibility to switch GNU math libraries order.

In the case below there are THREEfold quotes. Any help ?
~~~
configure_build_and_exe('fc_lapack', 'python setup.py --fc=gfortran --static --cmake-options="-DMATH_LIB_SEARCH_ORDER='ATLAS,SYSTEM_NATIVE,OPENBLAS'"')
~~~

Hi,
~~~
configure_build_and_exe('fc_lapack', 'python setup.py --fc=gfortran --static --cmake-options="-DMATH_LIB_SEARCH_ORDER=ATLAS,MKL"
~~~
gives
~~~
CMake Error at /home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/modules/math_libs.cmake:387 (set):
  Syntax error in cmake code at

    /home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/modules/math_libs.cmake:387

  when parsing string

    ${ATLAS,MKL_BLAS_INCLUDE_PATH_SUFFIXES}

  syntax error, unexpected cal_SYMBOL, expecting } (39)
Call Stack (most recent call first):
  /home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/modules/math_libs.cmake:406 (find_math_library)
  /home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/modules/math_libs.cmake:353 (find_service)
  /home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/modules/math_libs.cmake:474 (config_math_service)
  CMakeLists.txt:26 (include)
~~~

Also proper semicolons give error message:
~~~
    def test_fc_lapack_static():
>       configure_build_and_exe('fc_lapack', 'python setup.py --fc=gfortran --static --cmake-options="-DMATH_LIB_SEARCH_ORDER=ATLAS;MKL"')

../../test.py:198: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../test.py:112: in configure_build_and_exe
    stdout, stderr = exe(binary)
../../test.py:69: in exe
    universal_newlines=True).communicate()
/usr/lib64/python2.6/subprocess.py:642: in __init__
    errread, errwrite)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <subprocess.Popen object at 0x18ec910>, args = ['./bin/example'], executable = './bin/example', preexec_fn = None
close_fds = False, cwd = None, env = None, universal_newlines = True, startupinfo = None, creationflags = 0, shell = False, p2cread = None
p2cwrite = None, c2pread = 8, c2pwrite = 10, errread = 11, errwrite = 12

    def _execute_child(self, args, executable, preexec_fn, close_fds,
                       cwd, env, universal_newlines,
                       startupinfo, creationflags, shell,
                       p2cread, p2cwrite,
                       c2pread, c2pwrite,
                       errread, errwrite):
        """Execute program (POSIX version)"""
    
        if isinstance(args, types.StringTypes):
            args = [args]
        else:
            args = list(args)
    
        if shell:
            args = ["/bin/sh", "-c"] + args
            if executable:
                args[0] = executable
    
        if executable is None:
            executable = args[0]
    
        # For transferring possible exec failure from child to parent
        # The first char specifies the exception type: 0 means
        # OSError, 1 means some other error.
        errpipe_read, errpipe_write = os.pipe()
        try:
            try:
                self._set_cloexec_flag(errpipe_write)
    
                gc_was_enabled = gc.isenabled()
                # Disable gc to avoid bug where gc -> file_dealloc ->
                # write to stderr -> hang.  http://bugs.python.org/issue1336
                gc.disable()
                try:
                    self.pid = os.fork()
                except:
                    if gc_was_enabled:
                        gc.enable()
                    raise
                self._child_created = True
                if self.pid == 0:
                    # Child
                    try:
                        # Close parent's pipe ends
                        if p2cwrite is not None:
                            os.close(p2cwrite)
                        if c2pread is not None:
                            os.close(c2pread)
                        if errread is not None:
                            os.close(errread)
                        os.close(errpipe_read)
    
                        # Dup fds for child
                        if p2cread is not None:
                            os.dup2(p2cread, 0)
                        if c2pwrite is not None:
                            os.dup2(c2pwrite, 1)
                        if errwrite is not None:
                            os.dup2(errwrite, 2)
    
                        # Close pipe fds.  Make sure we don't close the same
                        # fd more than once, or standard fds.
                        if p2cread is not None and p2cread not in (0,):
                            os.close(p2cread)
                        if c2pwrite is not None and c2pwrite not in (p2cread, 1):
                            os.close(c2pwrite)
                        if errwrite is not None and errwrite not in (p2cread, c2pwrite, 2):
                            os.close(errwrite)
    
                        # Close all other fds, if asked for
                        if close_fds:
                            self._close_fds(but=errpipe_write)
    
                        if cwd is not None:
                            os.chdir(cwd)
    
                        if preexec_fn:
                            preexec_fn()
    
                        if env is None:
                            os.execvp(executable, args)
                        else:
                            os.execvpe(executable, args, env)
    
                    except:
                        exc_type, exc_value, tb = sys.exc_info()
                        # Save the traceback and attach it to the exception object
                        exc_lines = traceback.format_exception(exc_type,
                                                               exc_value,
                                                               tb)
                        exc_value.child_traceback = ''.join(exc_lines)
                        os.write(errpipe_write, pickle.dumps(exc_value))
    
                    # This exitcode won't be reported to applications, so it
                    # really doesn't matter what we return.
                    os._exit(255)
    
                # Parent
                if gc_was_enabled:
                    gc.enable()
            finally:
                # be sure the FD is closed no matter what
                os.close(errpipe_write)
    
            if p2cread is not None and p2cwrite is not None:
                os.close(p2cread)
            if c2pwrite is not None and c2pread is not None:
                os.close(c2pwrite)
            if errwrite is not None and errread is not None:
                os.close(errwrite)
    
            # Wait for exec to fail or succeed; possibly raising exception
            # Exception limited to 1M
            data = _eintr_retry_call(os.read, errpipe_read, 1048576)
        finally:
            # be sure the FD is closed no matter what
            os.close(errpipe_read)
    
        if data != "":
            _eintr_retry_call(os.waitpid, self.pid, 0)
            child_exception = pickle.loads(data)
            for fd in (p2cwrite, c2pread, errread):
                if fd is not None:
                    os.close(fd)
>           raise child_exception
E           OSError: [Errno 2] No such file or directory

/usr/lib64/python2.6/subprocess.py:1234: OSError
---------------------------------------------------------- Captured stderr call ----------------------------------------------------------
CMake Error: The source directory "/home/milias/Work/qch/software/software_projects/autocmake_devel/autocmake_miroi/test/fc_lapack/build-2015-08-10-13-29-42" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
/bin/sh: MKL: command not found
make: *** No targets specified and no makefile found.  Stop.
================================================== 1 failed, 13 passed in 63.24 seconds =============
~~~

--- a/test/test.py
+++ b/test/test.py
@@ -195,4 +195,4 @@ def test_fc_lapack():
 
 @no_windows
 def test_fc_lapack_static():
-    configure_build_and_exe('fc_lapack', 'python setup.py --fc=gfortran --static --cmake-options="-DMATH_LIB_SEARCH_ORDER=ATLAS"')
+    configure_build_and_exe('fc_lapack', 'python setup.py --fc=gfortran --static --cmake-options="-DMATH_LIB_SEARCH_ORDER=\'ATLAS;MKL\'"')

